### PR TITLE
Add sits_tsne() generic and methods for LTAE, TAE, TempCNN, MLP + plo…

### PR DESCRIPTION
**Pull request by alexcarssuncao**
### Summary of changes:

This pull request adds a new feature for visualizing feature embeddings from sits neural network models using t-SNE.

### ✅ New Functions:

- **`sits_tsne()`** (S3 generic)  
  - A generic interface for extracting embeddings from trained sits torch models and applying t-SNE dimensionality reduction.
  - Includes automatic duplicate filtering to avoid t-SNE errors.
  - Returns both the t-SNE result and aligned sample labels.

- **S3 Methods for `sits_tsne()`**  
  - **`sits_tsne.torch_model_ltae()`**: Extracts embeddings from the temporal encoder of LTAE models.  
  - **`sits_tsne.torch_model_tae()`**: Extracts embeddings from the temporal attention encoder of TAE models.  
  - **`sits_tsne.torch_model_tempcnn()`**: Uses the dense layer output after the TempCNN encoder as embeddings.  
  - **`sits_tsne.torch_model_mlp()`**: Extracts embeddings from the last hidden layer (before the output layer) of MLP models.

- **`plot.sits_tsne()`**  
  - A new S3 method for `plot()`.
  - Provides a simple ggplot2 scatterplot showing t-SNE projections colored by class labels.

### ✅ Usage Example:

```r
result <- sits_tsne(model, samples)
plot(result)

### CODE REVIEW

File: R/sits_plot.R [hunk 1]
```diff
@@ -2265,3 +2265,38 @@ plot.sits_cluster <- function(x, ...,
     )
     invisible(dend)
 }
+
+
+utils::globalVariables(c("X", "Y", "Class"))
```
Comment (M3nin0) [inline R/sits_plot.R]: Hi @alexcarssuncao,

Thanks a lot for your contribution.

Please, remove this `utils::globalVariables` call. You maybe added this to avoid errors with `devtools::check()` in line 2291

File: R/sits_plot.R [hunk 2]
```diff
@@ -2265,3 +2265,38 @@ plot.sits_cluster <- function(x, ...,
...
+plot.sits_tsne <- function(x, y, ...) {
+    .check_require_packages("ggplot2")
+
+    df_tsne <- data.frame(
+        X = x$tsne$Y[, 1],
+        Y = x$tsne$Y[, 2],
+        Class = x$labels
+    )
+
+    gp <- ggplot2::ggplot(df_tsne, ggplot2::aes(x = X, y = Y, color = Class)) +
```
Comment (M3nin0) [inline R/sits_plot.R]: To remove the `utils::globalVariables`. First, you need to update this code:

```
gp <- ggplot2::ggplot(df_tsne, ggplot2::aes(x = X, y = Y, color = Class)) + ...
```

In this case, you can reference `X`, `Y` and `Class` from the `df_tsne` using `.data`. For example:

```
gp <- ggplot2::ggplot(df_tsne, ggplot2::aes(x = .data[["X"]], y = .data[["Y"]], color = .data[["Class"]])) + ...
```

File: R/sits_mlp.R [hunk 3]
```diff
@@ -286,9 +286,16 @@ sits_mlp <- function(samples = NULL,
             colnames(values) <- labels
             values
         }
+        # Helper function to extract torch model (for tsne)
+        get_model <- function() {
+            model <- .torch_unserialize_model(serialized_model)
+            return(model)
+        }
+        # Attach the model as an attribute
+        attr(predict_fun, "get_model") <- get_model
```
Comment (M3nin0) [inline R/sits_mlp.R:295]: I think it’s better to avoid using `attr` in such cases. We can remove the `get_model` function and replace its usage in `sits_tsne` with the accessors available in `sits` to retrieve content from model closures.  

For example, in `sits_tsne.torch_model_tae`, instead of using `attr` + `get_model` as follows:

```
attr(model, "get_model")()
```

you can use the `.ml_model` accessor to retrieve the same model:

```r
.ml_model(model)[["model"]]
```

This produces the same result as `.torch_unserialize_model(serialized_model)` and also avoids the need to use `attr`.

File: R/sits_tsne.R [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+#' Applies t-SNE for TAE models using temporal encoder outputs as embeddings.
+#' @inheritParams sits_tsne
+#' @param remove_duplicates Logical. Should duplicated embeddings be removed before running t-SNE? Default is TRUE.
+#' @importFrom Rtsne Rtsne
+#' @export
+sits_tsne.torch_model_tae <- function(model, samples, remove_duplicates = TRUE, ...) {
+    # Check required packages
+    .check_require_packages(c("torch", "Rtsne"))
+    # Retrieve internal torch model
+    internal_model <- attr(model, "get_model")()
```
Comment (M3nin0) [inline R/sits_tsne.R]: Replace this `attr` + `function call` with `.ml_model(model)[["model"]]`

**Comment by M3nin0** [comment]
Hi @alexcarssuncao,

Thank you for your contributions. I left a few comments.

Could you also consider including a few test cases so we can validate the code and ensure it works as you expected?

Thanks,
Felipe Carlos

File: R/sits_tsne.R [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        embeddings_np <- as.array(embeddings)
+    })
+    # Original labels
+    labels <- samples$label
+    # Handle duplicates
+    if (remove_duplicates && any(duplicated(embeddings_np))) {
+        dup_idx <- !duplicated(embeddings_np)
+        embeddings_np <- embeddings_np[dup_idx, , drop = FALSE]
+        labels <- labels[dup_idx]
+        warning("Duplicate embeddings detected. Removed duplicates before running t-SNE.")
```
Comment (M3nin0) [inline R/sits_tsne.R]: Please replace this text to use the `messages` config. To do this, you need to:

1. Add your message to the `config_messages.yml` file with a name referencing the method (e.g., `sits_tsne_duplicated_embeddings`).
2. Call the message using `.config` (e.g., `warning(.conf("messages", "sits_tsne_duplicated_embeddings"))`).

File: R/sits_tsne.R [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+    # Get embeddings
+    embeddings_np <- NULL
+    internal_model$eval()
+    torch::with_no_grad({
+        spatial_out <- internal_model$spatial_encoder(x)
+        embeddings <- internal_model$temporal_attention_encoder(spatial_out)
+        embeddings_np <- as.array(embeddings)
+    })
+    # Original labels
+    labels <- samples$label
```
Comment (M3nin0) [inline R/sits_tsne.R]: Replace `$label` with `[["label"]]`. Example:

```
labels <- samples[["label"]]
```
This also applies to all cases where it is possible to replace `$` with `[[]]`. Let’s reserve `$` for R6 methods.

File: R/sits_tsne.R [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+#' @param samples A sits tibble with time series samples.
+#' @param ... Other parameters (passed to methods).
+#'
+#' @return A list with:
+#' \itemize{
+#'   \item \code{tsne}: The t-SNE result (object of class \code{"Rtsne"}).
+#'   \item \code{labels}: A character vector of labels, aligned with rows in \code{tsne$Y}.
+#' }
+#'
+#' @export
```
Comment (M3nin0) [inline R/sits_tsne.R:18]: I think it would be good to add a complete usage example as part of the function documentation.

File: DESCRIPTION [hunk 8]
```diff
@@ -64,7 +64,8 @@ Imports:
     tmap (>= 4.0),
     torch (>= 0.14.0),
     units,
-    utils
+    utils,
+    Rtsne
```
Comment (M3nin0) [inline DESCRIPTION:68]: Maybe, we can move this to `Suggests`

**Review by M3nin0 — CHANGES_REQUESTED** [review]
